### ISSUE TITLE
Proposal: Innovation Reviewer

### ISSUE THREAD

**Issue by njordhov** [blockstack/app-mining#192]
**What is the problem you are seeing? Please describe.**
A low-risk strategy for success in app mining is to base a new app on the design of an app that already have a high TMUI score. Several high scoring apps seem to have applied this strategy, including the top scorer this month.

**How is this problem misaligned with goals of app mining?**

It encourages a proliferation of similar apps instead of innovation, and directs funding to those that game the system rather than app developers with novel ideas or business models.

**What is the explicit recommendation you’re looking to propose?**

To encourage innovation and variety, create an Innovation reviewer that gives a low score for apps that are similar to existing Blockstack apps and a high score for apps that are distinct and novel. Apply this reviewer only for debuting apps in their first month (at least initially).

**Describe your long term considerations in proposing this change. Please include the ways you can predict this recommendation could go wrong and possible ways mitigate.**

It should lead to more diversity in apps, but may also result in early apps dominating a category, resulting in stagnation and users having less alternatives in popular categories. This can be mitigated by making the innovation score depend on the number of similar Blockstack apps.

**Additional context**
This proposal may help with related concerns, like the challenge of "clone apps".
 
https://github.com/blockstack/app-mining/issues/75
https://github.com/blockstack/app-mining/issues/113

**Comment by friedger** [comment]
@njordhov Do you have any ideas who to measure similarity? 
Is the similarity in app descriptions a good measurement ([using Tensorflow](https://cloud.google.com/solutions/machine-learning/analyzing-text-semantic-similarity-using-tensorflow-and-cloud-dataflow)?)? Can we measure similarity by using TMUI videos and some kind of AI?

**Comment by paradigma-cl** [comment]
The idea of using the Blockchain to build and manage network protocols was a very innovative application!
It is quite a challenge to evaluate innovation like in the App-mining program.  As it is today, most of the innovation has been transfering or coverting applications or functionalities that have been developed in another architecture to be accesible in a descentralized and Blockstack way.  That replica of what we have today is one level of innovation.  Many dapps have been a good example of this and they have received very good retribution for that effort.  Probably, the open source strategy for these developments have been excellent to speed up the learning and adoption curve of the Blockstack platform. One question arises, sustainability of the evolution of a dApp.
As in the book The Circle of Innovation by Tom Peters, there are serveral ideas that generate ideas, like Distance is dead, destruction is cool, you can't live without an eraser, we are all Michelangelos, all value comes from the professional services, the system is the solution, create waves of lust, branding, etc.....
Probably, another level of innovation could be the creation of real new ways solving the problems we are living in using these Circle of Innovation or other ways of innovating now enabled by the advantages of the Blockstack platform.  These type of innovations are more sofisticated in terms of it usage, and if they are really innovative, no other similar initiative can be easily detected.  In this case, it is not so clear the benefits of having an open source strategy or not for these innovations.  Depending on the business model, it can go one way or another to assure sustainability.

**Comment by friedger** [comment]
Example from November results:
Runkod (similar to no other app)
![Screenshot from 2019-11-29 11-52-02](https://user-images.githubusercontent.com/1449049/69864201-b3450880-129e-11ea-95b0-cff45eaf7e59.png)

Stackfolio (similar to Lannister)
![Screenshot from 2019-11-29 11-51-49](https://user-images.githubusercontent.com/1449049/69864214-b809bc80-129e-11ea-8e5a-25551a620a01.png)

Moodify (similar to Gekri, Nomie, Piara)
![Screenshot from 2019-11-29 11-55-27](https://user-images.githubusercontent.com/1449049/69864512-5e55c200-129f-11ea-9fe4-19f02bf1b05f.png)

**Comment by hstove** [comment]
I don't agree that building a distinct app indicates innovation, or that building something novel is even better in and of itself. If there are other apps out there that solve the same problem, and they have success, that's good validation that there's a market and need for that problem.

There is also the concern that determining exactly how 'similar/innovative' an app is would be totally subjective. If someone built another storage app, but had much better UX and some unique features, I would find that very innovative, even if it's "similar".

**Comment by friedger** [comment]
@hstove I see innovation/novelty as an indicator of a quality app.

> and they have success

How do you measure success? High rank in app mining means the algorithm thinks the app is successful, however, the relation between the algorithm view on success and the real world success still needs to be explained.

> There is also the concern that determining exactly how 'similar/innovative' an app is would be totally subjective.

We are here to solve problems ;-)
TMUI is a reviewer that measures totally subjective properties like usability, usefulness, credibility, and desirability. Let's define one for innovation!

> but had much better UX and some unique features

Yes, we both want less similar apps!

Currently, I have the feeling that good and distinct apps like runkod get lost in the noise of similar apps. (I take runkod's PH ranking - in a time when there are no incentives to buy upvotes - as an indicator of success.)
We 13 bookmark apps, 8 notes apps, 7 storage apps, 7 photo manager apps, 6 password manager apps, 5 photo editor apps, 4 daily notes apps,  4 blog apps, 3 2FA apps,..

**Comment by paradigma-cl** [comment]
It is difficult to measure innovation as it is a moving target.  Moving target in the app-mining initiative is associated with the opportunity (time) when a dapp was presented to the program.  In that moment, what the dapp is trying to accomplish, and the level of innovation, (or advantages) compared with other similar (if any) app-mining dapps.  
Another level of comparison, could be among the non app-mining apps available in the market.
If we consider as written  "if there are other apps out there that solve the same problem, and they have success, that's good validation that there's a market and need for that problem." then probably the dapps that have to be replicated from what it is already in the market, but using the Blockstack infrastructure.  Probably, this approach means that all innovation has been done, and only we have to apply the innovation provided by the Blockstack platform to different market proven succesfull applications.  This can be another category of innovation or simply a characteristic that indicates that it has a proven market, maybe also considering market reach, size, etc.
As technology is an enabler for new markets and possible new applications (for example, when initially mobile phones were able to access the Internet), it could result of new applications that were not succesful before the use of this enabling technology.  Most of the use of social network applications were accelerated, because of the massive use of mobile devices.  
So the question of what other innovations could be triggered by the use of the Blockstack platform and which of them is the App-mining program willing to measure and incentivise?  Is the app-mining program only interested in open source so the innovations can be reused in other dapps?  Or is the app-mining program interested in creating a eco-system that different innovative dapps that take advantage of the Blockstack platform are interacting and managing to be succesful by itself and by the help of the program? 
Looks like that to understand the reality of the level of innovations, success and market possibilities among the App-mining dapps, we should have some kind of monthly measurement.

**Comment by jeffdomke** [comment]
I agree, difficult to measure innovation. And some of the best innovation lies in taking an existing service, copying 90%, and innovating on something that could be subtle or even not in the app (like service, or distribution). And inversely, sometimes the most innovate ideas turn out to be pointless. Not sure we should reward pure innovation for innovation sake. 

Also not sure who this reviewer would be but open to ideas. FWIW I feel there are some entities out in the world that would be qualified to score on innovation.

**Comment by njordhov** [comment]
> Not sure we should reward pure innovation for innovation sake.

Keep in mind that the initial purpose of this reviewer is to avoid a situation in which the top scoring apps are just copycats of existing high scoring apps. The innovation reviewer will balance the incentives to make it also pay off to innovate rather than choose the low risk option of building on an already proven high scoring concept.

**Comment by paradigma-cl** [comment]
**To be safe we should look at the end of things.**
Depending how we see innovation can help us to develop applications that are enabling, useful, needed, valued among other qualities, we should think the incentives we are putting in place.

1. If we promote (positive scoring) the dapp development as Open Source, every month cycle probably the winner will be the one that has a better presentation and GUI in a certain desired solution (the more successful and known applications in other platforms, will have higher probabilities of being understood by the audience). But it would not indicate that the winning team has the technical capabilities entitled really to carry on developing that solution, as the code used has been obtained from others and the value added mainly will the marginal effort that has been made from the previous version of the code.  Nevertheless. this could a type of dApps that probably AppMining program wants to have as it is a way to train and get more developers participating in the challenge increasing the number of dApps.  Probably the ending dapp cycle will be the winner for each type of solution.  

2. If we promote more innovative/novel applications, every month cycle probably the winner could not be clearly predicted as there could be many factors that can be of success.  Obviously,still the already proven applications in other technological platforms will most of the time attract the attention as the audience know more about them.  The challenge, will always will be for the dapps that are trying to create a new type of applications, but in this case using the benefits of Blockstack technology.  Probably, initially there are not going to be big number of them. Right now, we still do not know which are these succesful new type of applications based in Blockstack. Do we want to encourage to find(develop) these new succesful applications?

3. Another way to innovate could be encouraging collaboration among the different teams. This could lead to a more sofisticated platform, or instead of having just one team to develop everything, the different team are working in different aspects of a major solution.  One can see different European Union collaboration funding programs in the area of technology to encourage different companies to work jointly in the development of a solution.

4. The Can't Be Evil premise is encouraging.  But we have to think of a way to audit the code without being open source or licensed.  Essentially, this flag was raised because of what has happened, affecting the whole world, so how in one moment to another, everybody will be good and love all humankind? Are we conducting a program to empower spiritual or moral values to the App-mining community developers? We have to think of a way to solve this issue of trust or certification of an application and build it.
